src/extension.ts

```diff
@@ -8,82 +8,89 @@ import * as path from 'path';
 
-		for ( var exclude in search_exclude_settings ) {
-			if( search_exclude_settings.hasOwnProperty( exclude ) ) {
-				exclude_files += exclude + ",";
-			}
-		}
+		const getNormalisedRelativeFilename = (wsfolder) => {
+			const fp = path.dirname(absolute_filename);
 
-		exclude_files += normalised_relative_filename_and_extension + "}";
+			let res = fp.slice(wsfolder.fsPath.length);
+			if (res.charAt(0) === '\\' || '/')  // linux or windows
+			    res = res.slice(1);
+			return path.join(res, filename).replace(/\\/g, '/'); // use only character / for glob usage
+  		};
 
-		var files = vscode.workspace.findFiles(filename_search, exclude_files, 100);
+		// the filename + search criteria to use for matching
+		let filename_search: string;
+		let relativePattern: vscode.RelativePattern;
+		if (mode !== ESwitchMode.All) {
+			const wsfolder = vscode.workspace.getWorkspaceFolder(absolute_uri);
+			if (!wsfolder) {
+				vscode.window.showErrorMessage(`No workspace for ${absolute_filename}`);
+				return;
+			}
+			// limiting search to same directory (in same workspace folder)
+			if (mode === ESwitchMode.Same_Dir)
+				filename_search = getNormalisedRelativeFilename(wsfolder.uri) + ".*";
+			// limiting search to same workspace folder
+			else // if (mode === ESwitchMode.Same_Workspace)
+				filename_search = "**/" + filename + ".*";
+			relativePattern = new vscode.RelativePattern(wsfolder, filename_search);
+		} else {
+			// search in all workspace folders
+			filename_search = "**/" + filename + ".*";
+		}
 
-		files.then((files) => {
+		// don't include files excluded from search in the workspace
+		// const search_exclude_settings = vscode.workspace.getConfiguration('search', absolute_uri).get('exclude');
+		const search_exclude_settings = vscode.workspace.getConfiguration('search', null).get('exclude');
+		const exclude_files = "{" +
+			Object.keys(search_exclude_settings)
+				.filter(key => search_exclude_settings[key]).toString() +
+			"}";
+
+		let findFiles: Thenable<vscode.Uri[]>;
+		if (relativePattern)
+			findFiles = vscode.workspace.findFiles(relativePattern, exclude_files, 100);
+		else
+			findFiles = vscode.workspace.findFiles(filename_search, exclude_files, 100);
+
+		findFiles.then((files: vscode.Uri[]) => {
 	        if (!files || files.length == 0) {
+				vscode.window.showInformationMessage('No files to switch found');
 				return;
 			}
 
-			// only want files whose name matches exactly
-			let exact_files = files.filter(file => path.basename(file.fsPath, path.extname(file.fsPath)) === filename);
+			// only want files whose name matches exactly and not the current one
+			const exact_files = files.filter(file =>
+					(file.fsPath !== absolute_filename) &&
+					path.basename(file.fsPath, path.extname(file.fsPath)) === filename);
+
 			if (!exact_files || exact_files.length == 0) {
+				vscode.window.showInformationMessage('No files to switch found');

```

Comment:
This will appear like that at the bottom (and auto disappear after a few time)
![image](https://user-images.githubusercontent.com/16486407/50557028-3b164180-0ce1-11e9-898b-3d3649315efd.png)

Reply:
Ah perfect good to know, thanks